src/webapp/components/template-selector/TemplateSelector.tsx

```diff
@@ -397,175 +425,190 @@ export const TemplateSelector = ({
-                                InputLabelProps={{ style: { color: "#494949" } }}
+                            <Select
+                                placeholder={i18n.t("Language")}
+                                onChange={onLanguageChange}
+                                options={availableLanguages}
+                                value={state.language ?? ""}
                             />
                         </div>
                     </div>
-                </>
-            )}
+                )}
 
-            {state.populate && state.type === "trackerPrograms" && userHasReadAccess && (
-                <div>
-                    <h4 className={classes.title}>
-                        {i18n.t("TEI and relationships enrollment by organisation unit type")}
-                    </h4>
+                {themeOptions.length > 0 && (
                     <div className={classes.row}>
                         <div className={classes.select}>
                             <Select
-                                value={state.relationshipsOuFilter}
-                                onChange={({ value }) =>
-                                    onRelationshipsOuFilterChange(value as RelationshipOrgUnitFilter)
-                                }
-                                options={[
-                                    {
-                                        label: i18n.t("Current user organisation units (data capture)"),
-                                        value: "CAPTURE",
-                                    },
-                                    {
-                                        label: i18n.t("Selected organisation units with their descendants"),
-                                        value: "DESCENDANTS",
-                                    },
-                                    { label: i18n.t("Only selected organisation units"), value: "SELECTED" },
-                                ]}
+                                placeholder={i18n.t("Theme")}
+                                onChange={onThemeChange}
+                                options={themeOptions}
+                                allowEmpty={true}
+                                emptyLabel={i18n.t("<No value>")}
+                                value={state.theme ?? ""}
                             />
                         </div>
                     </div>
+                )}
 
+                {isDataSet && state.templateType !== "custom" && (
+                    <FormControlLabel
+                        className={classes.checkbox}
+                        control={
+                            <Checkbox
+                                checked={state.splitDataEntryTabsBySection}
+                                onChange={toggleSplitDataEntryTabsBySection}
+                            />
+                        }
+                        label={i18n.t("Split data entry tabs by section")}
+                    />
+                )}
+
```

Comment:
[not to change] The return JSX expression of this component is huuuge (608-294). That's not your fault, you are just moving already existing parts. So this is a note for the future: whenever a component grows too much, let's abstract conceptual units and create new components. But again, we won't be doing it now, in the middle of a feature.